src/sagas/playground.ts

```diff
@@ -23,13 +25,19 @@ function* updateQueryString() {
   const chapter: number = yield select(
     (state: IState) => state.workspaces.playground.context.chapter
   );
+  const variant: Variant = yield select(
+    (state: IState) => state.workspaces.playground.context.variant
+  );
+
+  const languageUrlName: string = urlName(chapter, variant);
+
   const external: ExternalLibraryName = yield select(
     (state: IState) => state.workspaces.playground.externalLibrary
   );
   const execTime: number = yield select((state: IState) => state.workspaces.playground.execTime);
   const newQueryString: string = qs.stringify({
     prgrm: compressToEncodedURIComponent(codeString),
-    chap: chapter,
+    chap: languageUrlName,
```

Comment:
Should we encode the chapter and variant separately?

Reply:
Actually, the current method is simpler for validating the chapter - variant pair in the URL